Parse/src/main/java/com/parse/ParseInstallation.java

```diff
@@ -143,9 +143,7 @@ public String getInstallationId() {
       @Override
       public Task<Void> then(Task<Void> task) throws Exception {
         // Retry the fetch as a save operation because this Installation was deleted on the server.
-        // Do not attempt to resave an object if LDS is enabled, since changing objectId is not allowed.
```

Comment:
Only the `ParseException.OBJECT_NOT_FOUND` ([line 148](ParseException.OBJECT_NOT_FOUND)) can trigger the re-creating process. 
If just null out the objectId and save installation will not create another installation data. (It's weird, I'm still figuring out why)  
BTW, I think [`setObjectId()`](https://github.com/ParsePlatform/Parse-SDK-Android/blob/05795898c2cf63e78bedba6c9e20f569294685df/Parse/src/main/java/com/parse/ParseObject.java#L1232) should not be a public method, package private may be more appropriate.